File: java/src/test/java/problems1_100/IntegerToRomanTest.java
package problems1_100;

import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class IntegerToRomanTest extends SampledTest {
    @Test
    public void test() {
        for (IntegerToRoman solution : IntegerToRoman.values()) {
            for (IntegerToRomanSample sample : getSamples(IntegerToRomanSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.intToRoman(sample.input.num))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/IntegerToRomanTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class IntegerToRomanTest : SampledTest() {
    @Test
    fun test() =
        IntegerToRoman.entries.forEach { solution ->
            getSamples(Array<IntegerToRomanSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.intToRoman(input.num))
                        .isEqualTo(output)
                }
        }
}
